src/opm/input/eclipse/Schedule/Group/Group.cpp

```diff
@@ -656,7 +656,7 @@ bool Group::wellgroup() const {
 
 bool Group::addWell(const std::string& well_name) {
     if (!this->m_groups.empty())
-        throw std::logic_error("Groups can not mix group and well children. Trying to add well: " + well_name + " to group: " + this->name());
+        throw std::runtime_error("Groups can not mix group and well children. Trying to add well: " + well_name + " to group: " + this->name());
```

Comment:
Alas, one of those things that depend on the view. Is an eclipse file (part of a) program. Not worth spending time to change or rechange. Both are fine.

Reply:
I think this is clear if (1) illegal user input can put us here: then logic_error is wrong. However, if (2) a higher layer was supposed to eliminate this possibility, then being in this situation is a bug, and logic_error is right. Since there are no other checks for this anywhere else that I could find, this is the actual check for user input correctness and the case (1) applies.

On both this and the previous comment: the modified catch sequence in KeywordHandler.cpp should ensure that bugs that cause exceptions where there should be none still result in an "Internal error" message.